content-security-policy/script-src/script-src-strict_dynamic_double_policy_different_nonce.html

```diff
@@ -40,17 +40,16 @@
     </script>
 
     <script nonce='dummy'>
-        async_test(function(t) {
+        test(function(t) {
+            window.removeEventListener('securitypolicyviolation', cspViolationHandler);
+
             window.addEventListener('message', t.step_func(function(e) {
                 if (e.data === 'nonced-appendChild') {
                     t.done();
                 }
             }));
 
             window.addEventListener('securitypolicyviolation', t.step_func(function(violation) {
```

Comment:
Isn't this going to fire when the handler in the first block fires? You're attaching two listeners that expect opposite things. I don't understand how that will work.